compiler/sem.nim

```diff
@@ -165,6 +165,27 @@ proc commonType*(x, y: PType): PType =
         result = newType(k, r.owner)
         result.addSonSkipIntLit(r)
 
+proc procSymInCallExpr(n: PNode): PSym =
+  assert(n.kind in nkCallKinds)
+  var it = n
+  while it.kind != nkSym:
+    it = if it.kind == nkDotExpr: it[1]
+         else: it[0]
+  it.sym
+
+proc endsInNoReturn(n: PNode): bool =
+  # check if expr ends in raise exception or call of noreturn proc
+  var it = n
+  while it.kind in {nkStmtList, nkStmtListExpr} and it.len > 0: 
+    it = it.lastSon
+  result = it.kind == nkRaiseStmt or 
+    (it.kind in nkCallKinds and sfNoReturn in procSymInCallExpr(it).flags)
```

Comment:
The correct check is ``it.kind in nkCallKinds and it[0].kind == nkSym and sfNoReturn in it[0].sym.flags``

Reply:
done, as you say